src/main/java/net/minestom/server/item/ItemStack.java

```diff
@@ -195,6 +203,76 @@ public boolean equals(Object o) {
                 isSimilar((ItemStack) o) && ((ItemStack) o).getAmount() == getAmount();
     }
 
+    /**
+     * Checks if this item can be placed on the block.
+     * This should be enforced only for adventure mode players.
+     * @param block the block's namespaceID
+     * @return <code>true</code> if it can be placed, <code>false</code> otherwise
+     */
+    public boolean canPlaceOn(String block) {
+        return canPlaceOn.contains(block);
+    }
+
+    /**
+     * Adds the block to the list of blocks that
+     * this item can be placed on.
+     * @param block the block's namespaceID
+     */
+    public void addCanPlaceOn(String block) {
+        canPlaceOn.add(block);
+    }
+
+    /**
+     * Removes the block from the set of blocks that
+     * this item can be placed on.
+     * @param block the block's namespaceID
+     */
+    public void removeCanPlaceOn(String block) {
+        canPlaceOn.remove(block);
+    }
+
+    /**
+     * Gets the blocks that this item can be placed on
+     * @return an unmodifiable {@link Set} of blocks
+     */
+    public Set<String> getCanPlaceOn() {
```

Comment:
If you want to have a way to add/remove and retrieve the whole `Set`, you could as well have a single method returning a modifiable collection

Reply:
I introduced it because `net.minestom.server.utils.NBTUtils#saveDataIntoNBT` had to get the whole set, but kept the other methods to provide a shorter way of doing additions and removals, but I can change it if you want to.